src/com/xilinx/rapidwright/eco/FanOutOptimization.java

```diff
@@ -0,0 +1,238 @@
+import com.xilinx.rapidwright.edif.EDIFCell;
+import com.xilinx.rapidwright.edif.EDIFHierNet;
+import com.xilinx.rapidwright.edif.EDIFHierPortInst;
+import com.xilinx.rapidwright.placer.blockplacer.Point;
+import com.xilinx.rapidwright.util.Pair;
+import com.xilinx.rapidwright.util.Utils;
+
+/**
+ * Performs a fan out optimization of a high fan out net on a fully placed and
+ * routed design.
+ */
+public class FanOutOptimization {
+
+    public static Site getCentroidOfNet(Net net, Set<SiteTypeEnum> targetSiteTypes) {
+        List<Point> points = new ArrayList<>();
+        for (SitePinInst i : net.getPins()) {
+            points.add(new Point(i.getTile().getColumn(), i.getTile().getRow()));
+        }
+        Point centroid = KMeans.calculateCentroid(points);
+        Tile centroidTile = net.getSourceTile().getDevice().getTile(centroid.y, centroid.x);
+
+        // We need to snap to the closest site with the site type of interest from the
+        // centroid tile
+        Site closest = null;
+        int closetDist = Integer.MAX_VALUE;
+        int searchGridDim = 0;
+        while (closest == null) {
+            searchGridDim++;
+            for (int row = -searchGridDim; row < searchGridDim; row++) {
+                for (int col = -searchGridDim; col < searchGridDim; col++) {
+                    Tile neighbor = centroidTile.getTileNeighbor(col, row);
+                    if (neighbor != null) {
+                        for (Site s : neighbor.getSites()) {
+                            if (targetSiteTypes.contains(s.getSiteTypeEnum())) {
+                                int manDist = centroidTile.getManhattanDistance(neighbor);
+                                if (manDist < closetDist) {
+                                    closest = s;
+                                    closetDist = manDist;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return closest;
+    }
+
+    private static Pair<Site, BEL> findValidPlacementOption(Design design, Cell srcFF, Iterator<Site> itr) {
+        SiteInst si = srcFF.getSiteInst();
+        Net clk = si.getNetFromSiteWire(srcFF.getSiteWireNameFromLogicalPin("C"));
+        Net rst = si.getNetFromSiteWire(srcFF.getSiteWireNameFromLogicalPin("R"));
+        Net ce = si.getNetFromSiteWire(srcFF.getSiteWireNameFromLogicalPin("CE"));
+        while (itr.hasNext()) {
+            Site curr = itr.next();
+            SiteInst candidate = design.getSiteInst(curr.getName());
+            if (candidate == null) {
+                // Empty site, let's use it
+                return new Pair<Site, BEL>(curr, curr.getBEL("AFF"));
+            }
+            // Perhaps it is better to only use empty sites?
+//            BEL bel = h.getUnusedFlop(si, clk, ce, rst);
+//            if (bel != null) {
+//                return new Pair<Site, BEL>(curr, bel);
+//            }
```

Comment:
```suggestion
```

Reply:
I'm not certain that I want to give up on this alternative placement strategy yet.